### PR TITLE
Replace deprecated functions & add informative errors to `prepare_data`

### PR DESCRIPTION
**Pull request by efcaguab**
PACo uses `nullmodel` + `simulate` instead of `commsimulator` which is deprecated function in vegan. The new approach is also ~ 35% faster. Also, `prepare_data` has more informative errors when the number or names of species in the phylogenies and the interaction matrices don't match.

### CODE REVIEW

File: R/PACo.r [hunk 1]
```diff
@@ -23,9 +23,12 @@ PACo <- function(D, nperm=1000, seed=NA, method="r0")
    m2ss <- proc$ss
    pvalue <- 0
    if(!is.na(seed)) set.seed(seed)
+   # Create randomised matrices
+   null_model <- vegan::nullmodel (D$HP, method)
+   randomised_matrices <- simulate (null_model, nsim = nperm)
```
Comment (tpoisot) [inline R/PACo.r:28]: Where does `simulate` come from? Should it be `vegan::simulate` ?

Comment (efcaguab) [inline R/PACo.r:28]: simulate is in the stats package 
- Fer

**Comment by efcaguab** [comment]
I fixed a mistake in the prepare_data file. Other minor changes regarding exporting problems and the minimum version of vegan

**Comment by tpoisot** [comment]
Can you pull the current master branch? Some of these modifications act on code that was changed by @matthutchinson1 's code from a while back.

**Comment by efcaguab** [comment]
Uff should work now! :)

**Comment by tpoisot** [comment]
Done!

**Pull request merged**